### PR TITLE
chore: add Node.js 10

### PR DESCRIPTION
**Pull request by DanielRuf**

### CODE REVIEW

**Comment by kibertoad** [comment]
Unfortunately, there are no prebuilt Oracle binaries for Node 10 yet, so this won't pass the CI. Not sure if we want to run tests on incomplete set of dbs.

**Comment by DanielRuf** [comment]
Thanks for the hint, I'll check if CI passes and see if we can find a solution as 10 is the current stable version now.

**Comment by kibertoad** [comment]
@DanielRuf see https://github.com/oracle/node-oracledb/issues/901

**Comment by DanielRuf** [comment]
Ok so this can not be merged yet and I need further info from the maintainers so we can address the Node.js 10 issue(s).

**Comment by DanielRuf** [comment]
Looks better now. Can we retrigger Node.js 6? We finally caught real bugs on Node.js 10 (in our unit tests).

**Comment by kibertoad** [comment]
@DanielRuf close/open pr to rerun.

**Comment by kibertoad** [comment]
@DanielRuf can you explicitly bump oracle driver version in package.json?

**Comment by kibertoad** [comment]
@DanielRuf we didn't commit lock file before, are you sure we want to start doing it now?

**Comment by DanielRuf** [comment]
> we didn't commit lock file before, are you sure we want to start doing it now?

Check the failing builds, `npm ci` is used and needs synced lockfiles ;-)

**Comment by DanielRuf** [comment]
> @DanielRuf can you explicitly bump oracle driver version in package.json?

Which package do you mean in package.json?

**Comment by kibertoad** [comment]
My mistake, I thought we are using it as dev depency, but apparently only travis does.

**Comment by kibertoad** [comment]
@DanielRuf Both failures look like accidents, can you try rerunning the tests?

**Comment by DanielRuf** [comment]
> @DanielRuf Both failures look like accidents, can you try rerunning the tests?

Rerunning them will not fix them. 10 has actual failing tests (probably due to dependencies) and 6 has an issue finding the needed apt packages.

Completely different.

I'll have to do some testing and research, probably by enabling Travis CI on my fork.

**Comment by kibertoad** [comment]
@DanielRuf 
Are you sure about it?
https://travis-ci.org/tgriesser/knex/jobs/389603382 shows one test that I've definitely seen failing randomly before.

**Comment by DanielRuf** [comment]
We should keep an eye on it and try to resolve this with promises or rerunning the tst if it fails.

**Comment by DanielRuf** [comment]
randomely failing things are never good

**Comment by kibertoad** [comment]
Yeah. I think it makes sense to create a separate issue for this.

**Comment by kibertoad** [comment]
@DanielRuf Are you sure about package-lock being necessary, though? It always worked fine before, and all this PR changes is adding one more env.

**Comment by DanielRuf** [comment]
> @DanielRuf Are you sure about package-lock being necessary, though?

Yes, see https://travis-ci.org/tgriesser/knex/jobs/389597882#L817

**Comment by kibertoad** [comment]
I wonder where "out-of-date" package-lock.json comes in if there isn't supposed to be one. Maybe it has something to do with cache used?.. Additional "npm install" somewhere in travis script might help.
@elhigu Are you ok with adding package-lock for compatibility with newer npm, or should we investigate other options?

**Comment by DanielRuf** [comment]
> Package lock has not been necessary earlier, so I don't see why it would suddenly become obligatory for this PR.

Either someone changed it to npm ci on Travis or Travis changed the defaults. I'll change the install step to npm i and revert the change with the lockfile.

**Comment by kibertoad** [comment]
@DanielRuf no, it is most likely due to npm version change in newer node. If everything else fails, you could try locking node version at 10.2.1, I think it has old enough npm to just work.

**Comment by elhigu** [comment]
I dont think that locking node version is a viable solution, it will just break with node 11 and above. Overriding travis scripts nay work or maybe there is some flag for npm wich uses old behavior.

**Comment by DanielRuf** [comment]
It is not the default behaviour of Node.js 10 or npm to use npm ci ;-)

And lockimg the Node.js version is not a solution here.

I will add the `install` part to the Travis config.

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,5 @@
 yarn.lock
+package-lock.json
```
Comment (kibertoad) [inline .gitignore]: this change is meaningless, so probably makes sense to restore it to how it previously was to avoid noise in history

Comment (DanielRuf) [reply .gitignore]: It is not meaningless, please check all changes (I just moved it to the other lockfile ignore rule).

Comment (DanielRuf) [reply .gitignore]: This is better as the lockfile ignore rules are both at the top.

Comment (DanielRuf) [reply .gitignore]: Which is more consistent. To make it clear we can also add some comments.

Comment (kibertoad) [reply .gitignore]: Yeah, I meant the ordering. And yes, it is more consistent, you are right.

**Comment by elhigu** [comment]
it seems to work now 🎉 thank you!

**Comment by DanielRuf** [comment]
@elhigu @kibertoad see https://twitter.com/travisci/status/1006652447660609536

**Comment by DanielRuf** [comment]
So Travis CI changed the default behavior.

**Pull request merged**